frontend/dashboard/pages/CreateService/CreateService.tsx

```diff
@@ -1,153 +1,130 @@
-    setRepoName(newValue);
-    setRepoErrorMessage(null);
-  }, []);
+  const defaultSelectedOrgOrUser: string =
+    selectedContext === SelectedContextType.Self ? user.login : selectedContext;
+  const createAppRepo = async (createAppForm: CreateAppForm) => {
+    addRepo(
+      {
+        org: createAppForm.org,
+        repository: createAppForm.repoName,
+        datamodellingPreference: dataModellingPreference,
+      },
+      {
+        onSuccess: (): void => {
+          navigateToAppDevelopment(createAppForm.org, createAppForm.repoName);
+        },
+        onError: (error: AxiosError): void => {
+          const appNameAlreadyExists = error.response.status === ServerCodes.Conflict;
+          if (appNameAlreadyExists) {
+            setFormError(
+              (prevErrors): CreateAppForm => ({
+                ...prevErrors,
+                repoName: t('dashboard.app_already_exists'),
+              }),
+            );
+          }
+        },
+      },
+    );
+  };
 
-  const handleCreateService = async () => {
-    const isValid = validateInputs({
-      selectedOrgOrUser,
-      repoName,
-      t,
-      setRepoErrorMessage,
-      setOrgErrorMessage,
-    });
+  const handleCreateAppFormSubmit = async (
+    event: React.FormEvent<HTMLFormElement>,
+  ): Promise<void> => {
+    event.preventDefault();
 
-    if (isValid) {
-      setPageState(PageState.Creating);
+    const formData: FormData = new FormData(event.currentTarget);
 
-      await addRepo(
-        { org: selectedOrgOrUser, repository: repoName, datamodellingPreference: selectedFormat },
-        {
-          onSuccess: (repository) => {
-            window.location.assign(
-              applicationAboutPage({
-                org: repository.owner.login,
-                repo: repository.name,
-              }),
-            );
-          },
-          onError: (error: { response: { status: number } }) => {
-            if (error.response.status === ServerCodes.Conflict) {
-              setRepoErrorMessage(t('dashboard.app_already_exists'));
-            }
+    const createAppForm: CreateAppForm = {
+      org: formData.get('org') as string,
+      repoName: formData.get('repoName') as string,
+    };
 
-            setPageState(PageState.Idle);
-          },
-        },
-      );
+    const isFormValid: boolean = validateCreateAppForm(createAppForm);
+    if (isFormValid) {
+      await createAppRepo(createAppForm);
     }
   };
+
+  const validateCreateAppForm = (createAppForm: CreateAppForm): boolean => {
+    const { errorMessage: orgErrorMessage, isValid: isOrgValid } = validateRepoOwnerName(
+      createAppForm.org,
+    );
+    const { errorMessage: repoNameErrorMessage, isValid: isRepoNameValid } = validateRepoName(
+      createAppForm.repoName,
+    );
+
+    setFormError({
+      org: isOrgValid ? '' : orgErrorMessage,
+      repoName: isRepoNameValid ? '' : repoNameErrorMessage,
+    });
+
+    return isOrgValid && isRepoNameValid;
+  };
```

Comment:
Har du vurdert å gjøre dette også til en del av `useCreateAppFormValidation`?

Reply:
Det var egentlig fordi jeg synes koden ble mer tydelig å ha `setFormError` nærmere selve "skjema" og at det gjøres en `setError` i `onError` på selve mutation for å opprette et skjema. :)

Reply:
Skjønner, det la jeg ikke merke til. Da lar vi det være sånn. :+1: